src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -207,7 +207,7 @@ export class VegaBaseView {
     const vegaLoader = loader();
     const originalSanitize = vegaLoader.sanitize.bind(vegaLoader);
     vegaLoader.sanitize = async (uri, options) => {
-      if (uri.bypassToken === bypassToken) {
+      if (uri.bypassToken === bypassToken || this._externalUrl.isInternalUrl(uri)) {
```

Comment:
When I first saw this, I was going to ask why you added a new `isInternalUrl` function instead of just making this change:

```ts
if (uri.bypassToken === bypassToken || this._externalUrl.validateExternalUrl(uri) !== null) {
```

This would have the benefit of allowing external URLs that are configured in the global allow-list: https://www.elastic.co/guide/en/kibana/7.17/url-drilldown-settings-kb.html
The downside is that this global allow-list allows _all_ external URLs by default, so this would effectively break the existing `vis_type_vega.enableExternalUrls` setting, which _forbids_ all external URLs by default.
Just wanted to write this down to save future reviewers some time 😄 

@flash1293 has your team considered deprecating `vis_type_vega.enableExternalUrls` in favor of the global external URL service? This would provide a more consistent UX across Kibana.

----

Normally I'd prefer not to expose this `isInternalUrl` function, we don't really want consumers to be making decisions about whether a URL is allowed or not. However, given the existing behavior of Vega, I think this is OK/appropriate.

Also, I found this issue that this PR will resolve 🎉 , I'll link it: #67921

Reply:
Good point @jportner - we discussed this when introducing url validation, but decided against it because it would weaken the default security. I will bring it up again.